dedupe/dedupe.go

```diff
@@ -1,33 +1,30 @@
-package dedupe
+package dedup
 
 import (
-	"encoding/hex"
-	"fmt"
-	"hash"
+	"hash/maphash"
 	"sync"
 	"time"
 )
 
 // PacketDeduplicator is a structure that prevents processing of duplicate packets.
 // It keeps a record of seen packets and the time they were last seen.
 type PacketDeduplicator struct {
-	hasher       hash.Hash            // hasher is used for generating packet identifiers.
-	expiresAfter time.Duration        // expiresAfter defines the duration after which a seen packet record expires.
-	sync.RWMutex                      // RWMutex is used to protect the seen map from concurrent access.
-	seen         map[string]time.Time // seen maps a packet identifier to the last time it was seen.
+	expiresAfter time.Duration // expiresAfter defines the duration after which a seen packet record expires.
+	seed         maphash.Seed
+	sync.Mutex                 // protects the seen map from concurrent access.
+	seen         map[uint64]time.Time // seen maps a packet identifier to the last time it was seen.
 }
 
 // NewDeduplicator creates a new PacketDeduplicator with a given hasher and expiration duration for packet records.
 // It starts a background goroutine to periodically clean up expired packet records.
-func NewDeduplicator(hasher hash.Hash, expiresAfter time.Duration) *PacketDeduplicator {
-	pd := PacketDeduplicator{
-		seen:         make(map[string]time.Time),
-		hasher:       hasher,
+func NewDeduplicator(expiresAfter time.Duration) *PacketDeduplicator {
+	pd := &PacketDeduplicator{
+		seen:         map[uint64]time.Time{},
+		seed:         maphash.MakeSeed(),
 		expiresAfter: expiresAfter,
 	}
 	go func() {
-		for {
-			time.Sleep(expiresAfter)
+		for range time.NewTicker(time.Second * 10).C {
```

Comment:
Note the time is hardcoded here.  The sweep time doesn't matter for correctness, but when coupled with the expiration, you have to have much slower tests that use a lot more CPU.

This does mean that this code is effectively untested.